File: rb/features.rb
# MetMuseumCollection SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MetMuseumCollectionFeatures
  def self.make_feature(name)
    case name
    when "base"
      MetMuseumCollectionBaseFeature.new
    when "test"
      MetMuseumCollectionTestFeature.new
    else
      MetMuseumCollectionBaseFeature.new
    end
  end
end

File: py/metmuseumcollection_sdk/features.py
# MetMuseumCollection SDK feature factory

from metmuseumcollection_sdk.feature.base_feature import MetMuseumCollectionBaseFeature
from metmuseumcollection_sdk.feature.test_feature import MetMuseumCollectionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MetMuseumCollectionBaseFeature(),
        "test": lambda: MetMuseumCollectionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
